### PR TITLE
Proposed new notification module: BearyChat

### PR DESCRIPTION
**Pull request by tonyseek**
##### SUMMARY

BearyChat (https://bearychat.com) is a Slack alternative service. This is a notification module of BearyChat, just like the one of Slack.

##### ISSUE TYPE

 - New Module Pull Request

##### ADDITIONAL INFORMATION

The official organization of BearyChat is @bearyinnovative. I am not an employee of them and this module is unofficial.

### CODE REVIEW

**Comment by tonyseek** [comment]
Migrated from https://github.com/ansible/ansible-modules-extras/pull/1448

**Comment by shonenada** [comment]
shipit

**Comment by bcho** [comment]
shipit

**Review by bjolivot — CHANGES_REQUESTED** [review]
missclic

File: lib/ansible/modules/notification/bearychat.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    description:
+      - BearyChat WebHook URL. This authenticates you to the bearychat
+        service. It looks like
+        C(https://hook.bearychat.com/=ae2CF/incoming/e61bd5c57b164e04b11ac02e66f47f60).
+    required: true
+  text:
+    description:
+      - Message to send.
+    required: false
+    default: None
```
Comment (bjolivot) [inline lib/ansible/modules/notification/bearychat.py]: required: false and default: None are implicit, you can remove both

File: lib/ansible/modules/notification/bearychat.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                    msg="BearyChat doesn't have support for this kind of "
+                        "attachment image")
+            target_images.append(image)
+    return attachment
+
+
+def do_notify_bearychat(module, url, payload):
+    if not HAS_URLPARSE:
+        module.fail_json(msg='urlparse is not installed')
+    response, info = fetch_url(module, url, data=payload)
```
Comment (bjolivot) [inline lib/ansible/modules/notification/bearychat.py]: Why do you wait to fail ? I think you must fail in "main"

File: lib/ansible/modules/notification/bearychat.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+
+def main():
+    module = AnsibleModule(argument_spec={
+        'url': dict(type='str', required=True, no_log=True),
+        'text': dict(type='str', required=False, default=None),
+        'markdown': dict(default='yes', type='bool'),
+        'channel': dict(type='str', default=None),
+        'attachments': dict(type='list', required=False, default=None),
+    })
```
Comment (bjolivot) [inline lib/ansible/modules/notification/bearychat.py]: Required=false are implicit could you remove them ?
Default=None too in case of string

**Review by bjolivot — CHANGES_REQUESTED** [review]
Hello
Thank you for your module could you make some changes ?
Thank you

**Comment by tonyseek** [comment]
@bjolivot Thanks for your review. The changes have been revised.

**Comment by bjolivot** [comment]
shipit

**Comment by tonyseek** [comment]
ready_for_review

**Comment by yuanbohan** [comment]
shipit

**Comment by tonyseek** [comment]
bot_status

**Comment by tonyseek** [comment]
Why waiting on me... I have revised the description of pull request.

**Comment by bcho** [comment]
bot_broken

**Comment by gundalow** [comment]
`changes_requested_by: bjolivot` I believe @bjolivot needs to accept their GitHub review

**Review by bjolivot — APPROVED** [review]

**Comment by bjolivot** [comment]
bot_status

**Comment by bcho** [comment]
shipit

**Comment by bcho** [comment]
bot_status

**Comment by bcho** [comment]
OK, waiting another maintainer to shipit :shipit:

**Comment by willybarro** [comment]
shipit :shipit:

**Comment by tonyseek** [comment]
ready_for_review

**Comment by tonyseek** [comment]
bot_status

**Comment by gundalow** [comment]
closing/reopenign to trigger CI then I think we can get this merged

**Comment by tonyseek** [comment]
ready_for_review

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Merged into `devel` so this will be released in Ansible 2.4.0
Thanks for the PR and the reviews :)

**Pull request merged**